### PR TITLE
fix: Columns not passing properly from SQL Lab to Explore

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY

Columns were not passing properly from SQL Lab to Explore. This PR fixes that by passing the columns from `latestQuery.results.columns` into SaveQuery from SqlEditor.

### TESTING INSTRUCTIONS

- Go to SQL Lab and run a query with columns
- Click the carat `v` button on the save button and click Save Dataset
- Click Save & Explore
- Observe that the columns have passed to the dataset properly

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx [hunk 1]
```diff
@@ -151,6 +151,7 @@ export const SaveDatasetModal: FunctionComponent<SaveDatasetModalProps> = ({
   openWindow = true,
   formData = {},
 }) => {
+  console.log('FINDME SaveDatasetModal', datasource);
```
Comment (hughhhh) [inline superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx:154]: remove this

File: superset-frontend/src/SqlLab/components/SqlEditor/index.jsx [hunk 2]
```diff
@@ -663,6 +663,11 @@ class SqlEditor extends React.PureComponent {
       </Menu>
     );
 
+    const queryWithColumns = () => ({
+      ...qe,
+      columns: this.props.latestQuery?.results?.columns || [],
+    });
+    console.log('findme SqlEditor', qe, queryWithColumns());
```
Comment (hughhhh) [inline superset-frontend/src/SqlLab/components/SqlEditor/index.jsx:670]: remove this

File: superset-frontend/src/SqlLab/components/SaveQuery/index.tsx [hunk 3]
```diff
@@ -166,6 +167,7 @@ export default function SaveQuery({
     </Form>
   );
 
+  console.log('FINDME SaveQuery', query, queryPayload());
```
Comment (hughhhh) [inline superset-frontend/src/SqlLab/components/SaveQuery/index.tsx:170]: remove this

File: superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx (hunk 1 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx]: This was just debugging, it wasn't ready for review yet. I pushed up a "clean up" commit [`here`](https://github.com/apache/superset/pull/20975/commits/9f348bddec74224a808ac5ba88c7883e42f78c6a).

File: superset-frontend/src/SqlLab/components/SqlEditor/index.jsx (hunk 2 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/SqlLab/components/SqlEditor/index.jsx]: This was just debugging, it wasn't ready for review yet. I pushed up a "clean up" commit [`here`](https://github.com/apache/superset/pull/20975/commits/9f348bddec74224a808ac5ba88c7883e42f78c6a).

File: superset-frontend/src/SqlLab/components/SaveQuery/index.tsx (hunk 3 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/SqlLab/components/SaveQuery/index.tsx]: This was just debugging, it wasn't ready for review yet. I pushed up a "clean up" commit [`here`](https://github.com/apache/superset/pull/20975/commits/9f348bddec74224a808ac5ba88c7883e42f78c6a).

**Review by hughhhh — APPROVED** [review]
LGTM

**Review by AAfghahi — APPROVED** [review]

**Pull request merged**